cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -0,0 +1,112 @@
+
+#include "abstract_ray_runtime.h"
+
+#include <cassert>
+
+#include <ray/api.h>
+#include <ray/api/ray_config.h>
+#include <ray/api/ray_exception.h>
+#include "../util/address_helper.h"
+#include "local_mode_ray_runtime.h"
+
+namespace ray {
+namespace api {
+
+std::unique_ptr<AbstractRayRuntime> AbstractRayRuntime::ins_;
+std::once_flag AbstractRayRuntime::isInited_;
+
+AbstractRayRuntime &AbstractRayRuntime::DoInit(std::shared_ptr<RayConfig> config) {
+  std::call_once(isInited_, [config] {
+    if (config->runMode == RunMode::SINGLE_PROCESS) {
+      ins_.reset(new LocalModeRayRuntime(config));
+      GenerateBaseAddressOfCurrentLibrary();
+    } else {
+      throw RayException("Only single process mode supported now");
+    }
+  });
+
+  assert(ins_);
+  return *ins_;
+}
+
+AbstractRayRuntime &AbstractRayRuntime::GetInstance() {
+  RAY_CHECK(ins_ != nullptr);
+  return *ins_;
+}
+
+void AbstractRayRuntime::Put(std::shared_ptr<msgpack::sbuffer> data,
+                             const ObjectID &object_id) {
+  objectStore_->Put(object_id, data);
```

Comment:
```suggestion
  object_store_->Put(object_id, data);
```
Please follow google style guide everywhere